enterprise/backend/src/metabase_enterprise/data_editing/data_editing.clj

```diff
@@ -92,7 +92,7 @@
                                :when (not (isa? semantic_type :type/PK))]
                            [(keyword field-name)
                             (or (when (nil? coercion_strategy) identity)
-                                (data-editing.coerce/input-coercion-fn coercion_strategy)
+                                (:in (data-editing.coerce/coercion-fns coercion_strategy))
```

Comment:
I wonder if it's worth also having a test using the related hierarchy to "statically" check we don't have any missing entries?

Reply:
we're definitely missing some strategies which I've included as TODO in the code.
I also added a test but temporarily commented it out.

We should do that in a follow up.